modules/create_timepoint/jsx/create_timepoint.js

```diff
@@ -0,0 +1,355 @@
+        this.populateErrors({message: 'Server error.'});
+      }.bind(this),
+    });
+  }
+  /**
+   * @return {DOMRect}
+   */
+  render() {
+    // Waiting for async data to load.
+    if (!this.state.isLoaded) {
+      return <Loader/>;
+    }
+    // Include form errors.
+    const errors = this.state.errors;
+    // Include subproject select.
+    const subproject = this.state.form.display.subproject ? (
+      <SelectElement
+        id={'subproject'}
+        name={'subproject'}
+        ref={'subproject'}
+        label={'Subproject'}
+        value={this.state.form.value.subproject}
+        options={this.state.form.options.subproject}
+        onUserInput={this.setForm}
+        emptyOption={false}
+        disabled={false}
+        required={true}
+      />
+    ) : '';
+    // Include psc select.
+    const psc = this.state.form.display.psc ? (
+      <SelectElement
+        id={'psc'}
+        name={'psc'}
+        ref={'psc'}
+        label={'Site'}
+        value={this.state.form.value.psc}
+        options={this.state.form.options.psc}
+        onUserInput={this.setForm}
+        emptyOption={false}
+        disabled={false}
+        required={true}
+      />
+    ) : '';
+    // Include visit select.
+    const visit = this.state.form.display.visit ? (
+      <SelectElement
+        id={'visit'}
+        name={'visit'}
+        ref={'visit'}
+        label={'Visit label'}
+        value={this.state.form.value.visit}
+        options={this.state.form.options.visit}
+        onUserInput={this.setForm}
+        emptyOption={false}
+        disabled={false}
+        required={true}
+      />
+    ) : '';
+
+    if (!this.state.success) {
+      return (
+        <div>
+          <div>
+            <h3>Create Time Point</h3> <br/>
+            {errors}
+            <FormElement
+              name={'timepointInfo'}
+              fileUpload={false}
+              ref={'form'}
+              class={'form-group col-sm-12'}
+              onSubmit={this.handleSubmit}
+            >
+              <StaticElement
+                label={'DCCID'}
+                text={this.state.data.dccid}
+              />
+              {subproject}
+              {psc}
+              {visit}
+              <ButtonElement
+                label={'Create Time Point'}
+                type={'submit'}
+              />
+            </FormElement>
+          </div>
+        </div>
+      );
+    } else {
+      return (
+        <div>
+          <div>
+            <h3>New time point successfully registered.</h3>
+            <a href={'/' + this.state.url.params.candID}>
+              Click here to continue.
+            </a>
+          </div>
+        </div>
+      );
+    }
+  }
+}
+CreateTimepoint.propTypes = {
+  Module: PropTypes.string,
+  DataURL: PropTypes.string,
+};
+
+/**
+ * Render create_timepoint on page load.
+ */
+window.onload = function() {
+  const createTimepoint = (
+    <CreateTimepoint
+      Module='create_timepoint'
```

Comment:
I don't think this prop is necessary.

Reply:
@HenriRabalais I found that Module property included in previous modules using react. I guess it doesn't really serve a purpose.